utils/reserves.py

```diff
@@ -214,34 +216,77 @@ def getDataV2Pools(
         else:
             gccPrint(f'Error getting reserveInUSDC for pair {pool["LP"]}', "ERROR")
             totalStakedInUSDC = 0
+
     nonTriAPRs = []
 
     # Rewarder logic
     if pool["Rewarder"] != ZERO_ADDRESS:
-        rewardsPerBlockForItem = 0
-        doubleRewardUsdRatioForItem = 0
-        rewarderContractForItem = init_rewarder(pool["Rewarder"])
-        rewardDecimalsForItem = pool["RewarderTokenDecimals"]
-        rewardsPerBlockForItem = (
-            rewarderContractForItem.functions.tokenPerBlock().call()
-            / (10**rewardDecimalsForItem)
-        )
-        rewarderAddressForItem = rewarderContractForItem.functions.rewardToken().call()
-        print(f"Double rewards per block: {rewardsPerBlockForItem}")
-        doubleRewardUsdRatioForItem = getTokenUSDRatio(
-            w3, pool, rewarderAddressForItem, wnearUsdRatio, triUsdRatio
-        )
+        if pool["RewarderType"] == "Complex":
+            rewardsPerBlockForItem = 0
+            tokenUsdRatio = 0
+            rewarderContractForItem = init_n_rewarder(pool["Rewarder"])
+            numRewardTokens = rewarderContractForItem.functions.numRewardTokens().call()
 
-        nonTriAPRs.append(
-            {
-                "address": rewarderAddressForItem,
-                "apr": getAPR(
-                    doubleRewardUsdRatioForItem,
-                    rewardsPerBlockForItem,
-                    totalStakedInUSDC,
-                ),
-            }
-        )
+            for i in range(numRewardTokens):
+                rewardTokenAddressForItem = (
+                    rewarderContractForItem.functions.rewardTokens(i).call()
+                )
+                rewardTokenContractForItem = init_erc20(rewardTokenAddressForItem)
+                rewardDecimalsForItem = (
+                    rewardTokenContractForItem.functions.decimals().call()
+                )
+
+                rewardsPerBlockForItem = (
+                    rewarderContractForItem.functions.tokenPerBlock(i).call()
+                    / (10**rewardDecimalsForItem)
```

Comment:
this works, ya?

Reply:
ya should do, gets decimals onchain via erc20 view call,
tokenPerBlock is now uint256[] type compared to uint256 on the ComplexRewarder contract hence the index arg

got ruff python linter extension installed with auto format on save for vscode hence the bracket wraps and line breaks so I assume syntax is correct